### PR TITLE
Ensures that local identifiers are indexed

### LINKED ISSUE #304: Ensure search by local identifier works.

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Resolves #304

### CODE REVIEW

File: spec/controllers/catalog_controller_spec.rb [hunk 1]
```diff
@@ -35,6 +35,20 @@
       expect(facets["state_ssim"]).to eq ["complete", 1]
       expect(facets["display_subject_ssim"]).to eq [output.imported_metadata[0].subject.first, 1]
     end
+    it "can search by local identifiers" do
+      persister.save(resource: FactoryGirl.create_for_repository(:scanned_resource, local_identifier: "p3b593k91p"))
+
+      get :index, params: { q: "p3b593k91p" }
+      expect(assigns(:document_list).length).to eq 1
+    end
+    context 'with imported metadata' do
```
Comment (tpendragon) [inline spec/controllers/catalog_controller_spec.rb]: I don't think there's such a thing as an imported local identifier. @escowles ?

Comment (escowles) [reply spec/controllers/catalog_controller_spec.rb]: We're not mapping it now, but I think that records with local identifiers in 024 (like the Cicognara records) should have those imported into local_identifier.

Comment (tpendragon) [reply spec/controllers/catalog_controller_spec.rb]: Okay. This test still isn't right though - you need something like `FactoryGirl.create_for_repository(:scanned_resource, imported_metadata: [ImportedMetadata.new(local_identifier: "p3b593k91p")])`

Comment (jrgriffiniii) [reply spec/controllers/catalog_controller_spec.rb]: Understood, I shall proceeding by using a fixture from 8543429

**Comment by jrgriffiniii** [comment]
This is ready for another review

**Pull request merged**